bootable-jar/runtime/src/main/java/org/wildfly/core/jar/runtime/Server.java

```diff
@@ -0,0 +1,297 @@
+import java.nio.file.Path;
+import static java.security.AccessController.doPrivileged;
+import java.security.PrivilegedAction;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import org.jboss.as.controller.ControlledProcessState;
+import org.jboss.as.controller.ControlledProcessStateService;
+import org.jboss.as.controller.ModelControllerClientFactory;
+import org.jboss.as.controller.ProcessStateNotifier;
+import org.jboss.as.controller.client.ModelControllerClient;
+import org.jboss.as.controller.client.helpers.DelegatingModelControllerClient;
+import org.jboss.as.server.Bootstrap;
+import org.jboss.as.server.Main;
+import org.jboss.as.server.ServerEnvironment;
+import org.jboss.as.server.ServerService;
+import org.jboss.as.server.SystemExiter;
+import org.jboss.modules.Module;
+import org.jboss.modules.ModuleLoadException;
+import org.jboss.modules.ModuleLoader;
+import org.jboss.msc.service.ServiceActivator;
+import org.jboss.msc.service.ServiceContainer;
+import org.jboss.msc.service.ServiceController;
+import org.jboss.msc.value.Value;
+import org.wildfly.core.jar.runtime._private.BootableJarLogger;
+
+/**
+ * Bootable jar server. Inspired from Embedded Server API standalone server.
+ *
+ * @author jdenise
+ */
+final class Server {
+
+    interface ShutdownHandler {
+
+        void shutdown(int status);
+    }
+
+    private static final String MODULE_ID_VFS = "org.jboss.vfs";
+    private final PropertyChangeListener processStateListener;
+    private final String[] cmdargs;
+    private final Properties systemProps;
+    private final Map<String, String> systemEnv;
+    private final ModuleLoader moduleLoader;
+    private ServiceContainer serviceContainer;
+    private ControlledProcessState.State currentProcessState;
+    private ModelControllerClient modelControllerClient;
+    private ExecutorService executorService;
+    private ProcessStateNotifier processStateNotifier;
+    private final ShutdownHandler shutdownHandler;
+
+    private Server(String[] cmdargs, Properties systemProps,
+            Map<String, String> systemEnv, ModuleLoader moduleLoader,
+            ShutdownHandler shutdownHandler) {
+        this.cmdargs = cmdargs;
+        this.systemProps = systemProps;
+        this.systemEnv = systemEnv;
+        this.moduleLoader = moduleLoader;
+        this.shutdownHandler = shutdownHandler;
+
+        processStateListener = new PropertyChangeListener() {
+            @Override
+            public void propertyChange(PropertyChangeEvent evt) {
+                if ("currentState".equals(evt.getPropertyName())) {
+                    ControlledProcessState.State newState = (ControlledProcessState.State) evt.getNewValue();
+                    establishModelControllerClient(newState, false);
```

Comment:
The boolean argument should be yes instead of false. Doing so the internal variable that stores the currentProcessState gets updated upon an nitification.

Reply:
Ok.

Reply:
Just for the record, the requested changes were rejected due to https://issues.redhat.com/browse/WFCORE-4922